### PR TITLE
feat(background): store spent amount on payment to retrieve balance

### LINKED ISSUE #272: Show remaining balance of authorized amnt
When the WM extension is connected to a digital wallet, there is an amount of money/value that the user authorizes the extension to access from the user's connected wallet. 

The extension pop-up should accurately update the displayed remaining balance of the authorized total value.

- [x] https://github.com/interledger/web-monetization-extension/issues/376
- [ ] https://github.com/interledger/web-monetization-extension/issues/378
- [ ] https://github.com/interledger/web-monetization-extension/pull/386

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Closes #272.

## Changes proposed in this pull request

Saves `grantSpentDebitAmount` as "amountSpent" in storage, which we can later use to retrieve balance. To handle race conditions, we store the max of amount spent in the batch of multiple (possibly parallel) calls.

Extracted from https://github.com/interledger/web-monetization-extension/pull/379, but using throttling instead of debouncing.

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">2fa592980569aa17b90e318dbf144a65f0179c89</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/9759865145">Run #9759865145</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25545384619/artifacts/1659211842">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/25545384619/artifacts/1659211844">Download</a></td></tr>
</tbody>
</table>

File: src/background/services/openPayments.ts [hunk 1]
```diff
@@ -536,7 +537,16 @@ export class OpenPaymentsService {
           source: 'Web Monetization'
         }
       }
-    )
+    )) as OutgoingPaymentWithSpentAmounts
```
Comment (sidvishnoi) [inline src/background/services/openPayments.ts:540]: https://github.com/interledger/open-payments/pull/480#issuecomment-2202353862

**Comment by sidvishnoi** [comment]
image

**Review by raducristianpopa — APPROVED** [review]
Two comments:

**Pull request merged**